Test/dafny0/JavaUseRuntimeLib.dfy

```diff
@@ -0,0 +1,9 @@
+// RUN: %dafny "%s" /useRuntimeLib /out:%S/Output/DafnyConsole.jar /compileTarget:java > "%t"
```

Comment:
What does this look like in the new CLI?

Reply:
You can't do this with the new CLI, since there's no `--exclude-runtime` option in `dafny build`, and no `--compile` option in `dafny translate`

Reply:
One of the use cases for `dafny build` is to compile a library (rather than an executable). In that case, the user needs the choice, in java at least, whether to bundle the runtime into the new library or not. I'm not sure how each of these cases is accomplished with the new CLI and the changes you are implementing here.

Reply:
The intuitive meaning, to me, of --include-runtime for `dafny build` is whether to build the resulting library or executable with or without the runtime bundled in. Is that what your implementation accomplishes?

Reply:
For `dafny build` there shouldn't be an `--include-runtime`, because it includes the runtime by default There could be an `--exclude-runtime`, but I'm more inclined towards adding a `--compile` to `dafny translate`.

Reply:
We don't need --exclude-runtime, just --include-runtime:false for `dafny build`. That would solve the use case issue I have in mind. That build, internally, would have to be done with DafnyRuntime.jar on the classpath (and without the sources).

Reply:
Remy, if you concur with this use of --include-runtime for `dafny build`, I'm happy to put off implementing that behavior to a future PR. For Java at least, If the runtime library is included in multiple jars, Java just uses the class files from the first one -- so the only bad effects are library bloat and possible conflicts between versions.

Reply:
How do you feel about `--compile` for `dafny translate`, that way all your library use-cases are handled by `dafny translate`, instead of by both `dafny build` and `dafny translate` ?

Reply:
I feel that adding --compile to translate is just turning it into build; just like build with not deleting sources is close to translate. And if the two are to be merged, I'd prefer the name build. I understood the distinction to be:
translate -- produce source code without building it, for inspection or because one needs to incorporate the source into a build system in a way that is more complex than `dafny build`. If we want we could also write out a csproj/gradle/Makefile/antfile or whatever to help the user build it.
build -- produce a library or executable, incorporating external code if desired, but ready to be used; that is, the user does not have to figure out how to build the Dafny translation artifacts into the library/executable
run -- a way to quickly translate-build-run programs with simple build systems.

And I still like that distinction.
- I like the idea of auto-detecting the platform
- Also like producing platform specific binaries as an option (i.e.,. as an alternate but not in place of an executable jar for java)
- On your third bullet: good with that as well, given the user has an option to pick the underlying mechanism
- And I'm all for a good library as well.

As for your comments on library use case. There are simple use cases as well. Just compile and package up some Dafny functionality into a library one uses locally (not distributed). And not nearly all Java uses require Gradle, .Net or maven. I would not want simple uses of Dafny to require users to be setup with build environments.